examples/http-certification/assets/src/backend/src/lib.rs

```diff
@@ -0,0 +1,175 @@
+use base64::engine::general_purpose::STANDARD as BASE64;
+use base64::Engine;
+use ic_asset_certification::{Asset, AssetConfig, AssetFallbackConfig, AssetRouter};
+use ic_cdk::{
+    api::{data_certificate, set_certified_data},
+    *,
+};
+use ic_certification::HashTree;
+use ic_http_certification::{HeaderField, HttpCertificationTree, HttpRequest, HttpResponse};
+use include_dir::{include_dir, Dir};
+use serde::Serialize;
+use std::cell::RefCell;
+
+// Public methods
+#[init]
+fn init() {
+    certify_all_assets();
+}
+
+#[post_upgrade]
+fn post_upgrade() {
+    init();
+}
+
+#[query]
+fn http_request(req: HttpRequest) -> HttpResponse {
+    serve_asset(&req)
+}
+
+thread_local! {
+    static HTTP_TREE: RefCell<HttpCertificationTree> = RefCell::new(HttpCertificationTree::default());
+    static ASSET_ROUTER: RefCell<AssetRouter<'static>> = RefCell::new(AssetRouter::default());
```

Comment:
if you wrap the `HTTP_TREE` in a `Rc` you should be able to add it to the router with a builder pattern, which would be nice imo to avoid doing `HTTP_TREE.with` & `ASSET_ROUTER.with`

Reply:
I agree. I'll give it a try and follow up with another PR